### PR TITLE
`splitArray` utility

### PR DESCRIPTION
**Pull request by giraugh**
Adds utility for splitting arrays in a similar fashion to `String.prototype.split`

### CODE REVIEW

**Review by GRA0007 — APPROVED** [review]
Would be good to get some more tests of the edge cases

File: lib/arrays/splitArray.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  return groups
+}
+
+if (import.meta.vitest) {
+  const { expect, it } = import.meta.vitest
+
+  it('works for example 1', () => {
+    const runs = splitArray([1, 3, 5, 6, 7, 9, 11], (x) => x % 2 === 0) 
+    expect(runs).toEqual([[1, 3, 5], [7, 9, 11]])
+  })
```
Comment (GRA0007) [inline lib/arrays/splitArray.ts:39]: Here are some you can use

```suggestion
  it('works for example 1', () => {
    const runs = splitArray([1, 3, 5, 6, 7, 9, 11], (x) => x % 2 === 0) 
    expect(runs).toEqual([[1, 3, 5], [7, 9, 11]])
  })
  
  it('handles an empty array', () => {
    const runs = splitArray([], (x) => x === 0) 
    expect(runs).toEqual([])
  })
  
  it('works if predicate is always true', () => {
    const runs = splitArray(['a', 'b', 'c'], () => true) 
    expect(runs).toEqual([[], [], [], []])
  })
```
